### PR TITLE
 Added support for user define custom actions

### PR DESCRIPTION
**Pull request by AndreasArvidsson**

### CODE REVIEW

File: src/actions/CommandAction.ts [hunk 1]
```diff
@@ -22,18 +22,22 @@ export default class CommandAction implements Action {
...
     { insideOutsideType: "inside" },
   ];
+  private command: string;
   private ensureSingleEditor: boolean;
 
   constructor(
     private graph: Graph,
-    private command: string,
-    { ensureSingleEditor = false } = {}
+    { command = "", ensureSingleEditor = false } = {}
```
Comment (pokey) [inline src/actions/CommandAction.ts:30]: I'd be tempted to use `null` instead of empty string here

File: src/actions/CommandAction.ts [hunk 2]
```diff
@@ -69,7 +73,7 @@ export default class CommandAction implements Action {
 
   async run(
     [targets]: [TypedSelection[]],
-    { showDecorations = true } = {}
+    { command = "", showDecorations = true } = {}
```
Comment (pokey) [inline src/actions/CommandAction.ts:76]: As above, I'd probably just make this one optional or nullable and then use `??` below

File: src/actions/index.ts [hunk 3]
```diff
@@ -59,6 +60,7 @@ class Actions implements ActionRecord {
   replaceWithTarget = new Bring(this.graph);
   reverseTargets = new Reverse(this.graph);
   rewrapWithPairedDelimiter = new Rewrap(this.graph);
+  runCommandOnSelection = new RunCommandOnSelection(this.graph);
```
Comment (pokey) [inline src/actions/index.ts:63]: I believe you should add this one to the `Action` type.  Would be nice if there were some way to make it error when you forget to do so 🤔

Also as mentioned in other PR, I think I'd tweak this name

**Review by pokey — COMMENTED** [review]
Looks good with some minor comments

**Review by pokey — COMMENTED** [review]
Looks good, but could use a couple tests for the new `executeCommand`, esp exercising a couple args such as `restoreSelection`

**Comment by AndreasArvidsson** [comment]
Paste and copy use restore selection so already tested that way. I can add some more if you like?

**Comment by AndreasArvidsson** [comment]
I added a test for a custom command. More than that I don't know how to do. We do not support different arguments in custom command csv yet and none of the makeshift actions do anything that our test recorder can use.

**Review by pokey — APPROVED** [review]

**Pull request merged**